src/TileSlider/TileSlider.tsx

```diff
@@ -217,7 +231,7 @@ const TileSlider = <T extends unknown>({
       document.addEventListener('touchend', handleTouchEnd);
       document.addEventListener('touchcancel', handleTouchCancel);
     },
-    [minimalTouchMovement, slide],
+    [minimalTouchMovement, slide, onSwipeStart, onSwipeEnd],
```

Comment:
This possibly is a risky change since the touch listeners rely on the touch sequence in order to work. I'm not sure what happens when the `useCallback` gets updated halfway through the touch sequence. 

When the `onSwipeStart` or `onSwipeEnd` callback changes from the parent component, this rebinds the `onTouchStart` event listener.

You can ignore this if this has no effect on the touch behavior.

Otherwise, we could use the [not-so-recommended](https://reactjs.org/docs/hooks-faq.html#how-to-read-an-often-changing-value-from-usecallback) useEventCallback pattern.

```ts
const useEventCallback = (fn) => {
  const fnRef = useRef(fn); // initialize a ref holding the function
  fnRef.current = fn; // update the fnRef each render with the latest fn
  
  // return a non changing function which always calls the latest fn
  return useCallback((...args) => typeof fnRef.current === 'function' ? fnRef.current(...args) : null, []); 
};

// in render
const swipeStartHandler = useEventCallback(onSwipeStart);
const swipeEndHandler = useEventCallback(onSwipeEnd);
const slideEndHandler = useEventCallback(onSlideEnd);
```

You can safely ignore these handlers from the dependencies and call them from the touch handler.